plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -184,6 +184,7 @@ void checkAndSetPorts(PortProber portProber) throws CoreException {
 
     if (portProber.isPortInUse(adminPort)) {
```

Comment:
A couple of things:
  1. Can you consult a server attribute to disable this auto-correction.  I could see us having a tick box in the somewhere in the server UI for _automatically select ports to avoid conflicts_.
  2. If we're doing auto-correcting the admin port, then should we also auto-correct the `api_port` too?  Otherwise our launch will still fail if it has a conflict.  It's always opened.